src/SFA.DAS.AssessorService.Application.Api.Client/Azure/AzureApiClient.cs

```diff
@@ -0,0 +1,243 @@
+                user.Subscriptions.AddRange(subscriptions);
+            }
+
+            return user;
+        }
+
+        public async Task<AzureUser> GetUserDetailsByEmail(string email, bool includeSubscriptions = false)
+        {
+            AzureUser user;
+            using (var httpRequest = new HttpRequestMessage(HttpMethod.Get, $"/users?api-version=2017-03-01&expandGroups=false&$top=1&$filter=email eq '{email}'"))
+            {
+                var response = await RequestAndDeserialiseAsync<AzureUserResponse>(httpRequest, "Could not get User");
+                user = response.Users.FirstOrDefault();
+            }
+
+            if (user != null && user.AzureId != null && includeSubscriptions)
+            {
+                var subscriptions = await GetSubscriptionsForUser(user.Id);
+                user.Subscriptions.AddRange(subscriptions);
+            }
+
+            return user;
+        }
+
+        private async Task<IEnumerable<AzureSubscription>> GetSubscriptionsForUser(string userId)
+        {
+            using (var httpRequest = new HttpRequestMessage(HttpMethod.Get, $"/users/{userId}/subscriptions?api-version=2017-03-01"))
+            {
+                var response = await RequestAndDeserialiseAsync<AzureSubscriptionResponse>(httpRequest, "Could not get Subscriptions for User");
+                return response.Subscriptions;
+            }            
+        }
+
+        public async Task<AzureUser> CreateUser(string ukprn, string username)
+        {
+            var userId = Guid.NewGuid();
+            var organisation = await _organisationsApiClient.Get(ukprn);
+            var contact = await _contactsApiClient.GetByUsername(username);
+
+            if (await GetUserDetailsByUkprn(ukprn) != null) throw new Exceptions.EntityAlreadyExistsException($"Access is already enabled for ukprn {ukprn}");
+            else if (await GetUserDetailsByEmail(contact.Email) != null) throw new Exceptions.EntityAlreadyExistsException($"Access is already enabled for username {username}");
+
+            var request = new CreateAzureUserRequest
+            {
+                FirstName = organisation.EndPointAssessorName,
+                LastName = "EPAO",
+                Email = contact.Email,
+                Note = ukprn,
+                Password = "3pa0Pa55w0rd!"
+            };
+
+            AzureUser user;
+            using (var httpRequest = new HttpRequestMessage(HttpMethod.Put, $"/users/{userId}?api-version=2017-03-01"))
+            {
+                user = await PostPutRequestWithResponse<CreateAzureUserRequest, AzureUser>(httpRequest, request);
+            }
+
+            if (user.AzureId != null)
+            {
+                var subscription = await SubscribeUserToProduct(user.Id, _productId);
+                user.Subscriptions.Add(subscription);
+            }
+
+            //await _organisationsApiClient.Update(new UpdateOrganisationRequest
```

Comment:
Picky point:- Commented out code for me should be assigned to the source control system
might be worthwhile in group discussion.